tachyon/zk/plonk/halo2/argument.h

```diff
@@ -30,36 +30,30 @@ class Argument {
       const std::vector<Evals>* fixed_columns,
       const std::vector<Poly>* fixed_polys,
       const ConstraintSystem<F>& constraint_system,
-      std::vector<std::vector<std::vector<F>>>&& instance_columns_vec) {
+      std::vector<std::vector<Evals>>&& instance_columns_vec) {
     size_t num_circuits = circuits.size();
 
     // Generate instance polynomial and write it to transcript.
     std::vector<std::vector<Poly>> instance_polys_vec =
         GenerateInstancePolys(prover, instance_columns_vec);
 
-    // Generate instance columns
+    // Append leading zeros to each column of |instance_columns_vec|.
     size_t n = prover->pcs().N();
-    std::vector<std::vector<Evals>> instance_columns_vec_with_leading_zero =
-        base::Map(instance_columns_vec,
-                  [n](std::vector<std::vector<F>>& instances_vec) {
-                    return base::Map(instances_vec,
-                                     [n](std::vector<F>& instances) {
-                                       // Append leading zeros to |instances|.
-                                       instances.resize(n, F::Zero());
-                                       return Evals(std::move(instances));
-                                     });
-                  });
+    for (size_t i = 0; i < num_circuits; ++i) {
+      for (Evals& instance_column : instance_columns_vec[i]) {
+        instance_column.evaluations().resize(n, F::Zero());
```

Comment:
```suggestion
        instance_column.evaluations().resize(n);
```
+ shouldn't this change go with the previous commit?